####-sparse-observable.md

```diff
@@ -0,0 +1,518 @@
+(See the "Questions" section for greater discussion of this.)
+For example: matrix multiplication of two operators of the form $\sum_i c_i A_i$ for an operator alphabet $`\{A_i\}`$ containing the Pauli operators and projectors onto their eigenstates is not closed in the same way; a projector multiplied by a different Pauli produces a term like $\lvert 0\rangle\langle 1\rvert$, which requires the sum of two alphabet terms to express.
+This leads to the same exponential explosion problems as `SparsePauliOp` has in representing projectors.
+This happens because the alphabet is formed of Hermitian operators (even though the complete observable need not be), but matrix multiplication does not preserve hermicity.
+
+This can in theory be avoided by storing an explicit $2\times2$ matrix of coefficients for single-qubit term.
+For observables whose terms all involve only one or two qubits, the memory impacts are potentially acceptable, but the constant-factor scaling as the number of non-identity qubits increases is far poorer overall.
+Asymptotically, doing this would have a memory usage of 36 bytes per non-identity qubit + 8 bytes per term, compared to 5 bytes per non-identity qubit + 16 bytes per term, and would correspondingly be more expensive to compute with and to decompose to "measurement" operations.
+
+Supporting things like this efficiently in the observable is also perhaps misleading for the user; the _measurement terms_ needed to support things like $\lvert 0\rangle\langle 1\rvert$ still involve the exponential explosion (they go like $X + iY$), which could easily lead to a user getting needing / being charged for exponentially scaling runtime from an operator that appeared benign.
+
+
+## Questions
+
+### Should `SparseObservable` allow parametric observables?
+
+The definition given of `SparseObservables` limits the coefficients to be double-precision complex floats.
+`SparsePauliOp` technically allows observables to be parametrised in terms of Qiskit's symbolic `Parameter` class (or really any Python object).
+
+Should `SparseObservable` also support these uses?
+
+Pros of support:
+
+* More flexibility in what can be represented in the operators.
+
+Cons:
+
+* Operations on the type will be more complex, and the general case will need to hold the Python GIL in Rust space to act on the observables.
+* More complex code is needed from Rust space to handle mathematical manipulations efficiently in the happy path of "no Python-space parametrisation".
+
+Jake: personally I'd avoid this unless we have a really strong compelling use-case for giving it first-class support.
+A user can always work around this simply by splitting the terms of their sum into different broadcast axes in the estimator pub, then calculating the sums themselves, which gives them far more freedom.
```

Comment:
I haven't seen strong use cases requiring parameterized observables, so I am fine leaving this out.

Reply:
There are some use-cases around parameterized observables but I agree that I would not classify them as "strong", especially since there are alternative means by which to achieve the same end-result.

That said, I wonder if it will be possible to have some Python-level utilities (or maybe alternative classes) that could make working with parameterized observables a little bit easier. I am not sure what exactly that would look like, but am wondering if @jakelishman could comment on the feasibility of something like this.